### PR TITLE
JDK-8301068: Replace NULL with nullptr in share/jvmci/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/jvmci/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301068](https://bugs.openjdk.org/browse/JDK-8301068): Replace NULL with nullptr in share/jvmci/

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tom Rodriguez](https://openjdk.org/census#never) (@tkrodriguez - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12183/head:pull/12183` \
`$ git checkout pull/12183`

Update a local copy of the PR: \
`$ git checkout pull/12183` \
`$ git pull https://git.openjdk.org/jdk pull/12183/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12183`

View PR using the GUI difftool: \
`$ git pr show -t 12183`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12183.diff">https://git.openjdk.org/jdk/pull/12183.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:1295]: Fix

File: src/hotspot/share/jvmci/jvmciCompilerToVMInit.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciCompilerToVMInit.cpp:155]: Align

File: src/hotspot/share/jvmci/jvmciEnv.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1692]: Align

File: src/hotspot/share/jvmci/jvmciEnv.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.hpp:50]: Align

File: src/hotspot/share/jvmci/jvmciEnv.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciEnv.hpp:67]: null

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:489]: Align

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:151]: Align

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:294]: Align

File: src/hotspot/share/jvmci/jvmciRuntime.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:274]: str

File: src/hotspot/share/jvmci/jvmciRuntime.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:336]: Comparison use nullptr

File: src/hotspot/share/jvmci/jvmciRuntime.cpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:587]: null

File: src/hotspot/share/jvmci/jvmci_globals.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/jvmci_globals.hpp:132]: Align

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:33]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:38]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:44]: null

File: src/hotspot/share/jvmci/vmStructs_jvmci.hpp
Comment (jdksjolen) [inline src/hotspot/share/jvmci/vmStructs_jvmci.hpp:50]: null

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:1244]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciEnv.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1475]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciEnv.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciEnv.cpp:1462]: `nullptr` in message.

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:202]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:236]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:243]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:249]: Align `\`

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:269]: Right. This is exception case you listed: expression in comment.

File: src/hotspot/share/jvmci/jvmciJavaClasses.cpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.cpp:624]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciJavaClasses.hpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciJavaClasses.hpp:394]: Should we align `\` in previous lines?

File: src/hotspot/share/jvmci/jvmciObject.hpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciObject.hpp:68]: `nullptr` in message

File: src/hotspot/share/jvmci/jvmciObject.hpp
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciObject.hpp:77]: `nullptr` in message

**Review by vnkozlov — COMMENTED** [review]
Few missed `nullptr` in messages.

**Comment by jdksjolen** [comment]
Thanks @vnkozlov, I've fixed the issues you found!

**Comment by vnkozlov** [comment]
You need second review from JVMCI experts. @dougxc or @tkrodriguez please look.

**Review by tkrodriguez — APPROVED** [review]

**Comment by jdksjolen** [comment]
Thank you, integrating.

/integrate

**Pull request closed without merging**
